mobile/src/components/course/course.tsx

```diff
@@ -1,39 +1,111 @@
+import { DataStatus, PermissionKey } from '~/common/enums/enums';
+import {
+  CategoryGetAllItemResponseDto,
+  CourseGetResponseDto,
+  CourseUpdateCategoryRequestDto,
+} from '~/common/types/types';
 import {
   BackButton,
   Content,
+  Dropdown,
+  Icon,
   Image,
+  Pressable,
   ScrollView,
   Spinner,
   Text,
   View,
 } from '~/components/common/common';
-import { getImageUri } from '~/helpers/helpers';
+import { checkHasPermission, getImageUri } from '~/helpers/helpers';
 import {
+  useAppDispatch,
+  useAppForm,
   useAppNavigate,
   useAppSelector,
+  useCallback,
   useEffect,
+  useFocusEffect,
+  useState,
   useWindowDimensions,
 } from '~/hooks/hooks';
+import { coursesActions } from '~/store/actions';
+import { courseUpdateCategory as courseUpdateCategoryValidationSchema } from '~/validation-schemas/validation-schemas';
 
+import { getDefaultUpdateCourseCategoryPayload } from './common';
+import { Category } from './components/components';
 import { styles } from './styles';
 
 const Course: FC = () => {
+  const [editMode, setEditMode] = useState(false);
   const navigation = useAppNavigate();
   const { width } = useWindowDimensions();
-  const { course, dataStatus } = useAppSelector((state) => state.courses);
+  const dispatch = useAppDispatch();
+  const user = useAppSelector((state) => state.auth.user);
+
+  const { course, dataStatus, categories } = useAppSelector(
+    (state) => state.courses,
+  );
   const dataCourse = course as CourseGetResponseDto;
+  const currentCategory = course?.category;
+
+  const handlePressEditIcon = (): void => {
+    setEditMode((prev) => !prev);
+  };
+
+  const { control, handleSubmit, reset } = useAppForm<
+    CourseUpdateCategoryRequestDto | CategoryGetAllItemResponseDto
+  >({
+    defaultValues: getDefaultUpdateCourseCategoryPayload(
+      currentCategory?.id ?? 0,
+    ),
+    validationSchema: courseUpdateCategoryValidationSchema,
+  });
+
+  const hasEditCategoryPermission = checkHasPermission({
+    permissionKeys: [PermissionKey.MANAGE_CATEGORIES],
+    userPermissions: user?.permissions ?? [],
+  });
+
+  const handleSelectNewCategory = (payload: number): void => {
+    dispatch(
+      coursesActions.updateCategory({
+        courseId: (course as CourseGetResponseDto).id,
+        newCategoryId: payload,
+      }),
+    );
+
+    handlePressEditIcon();
+  };
 
   useEffect(() => {
     navigation.setOptions({
       headerLeft: () => <BackButton onPress={navigation.goBack} />,
+      headerRight: () =>
+        hasEditCategoryPermission && (
+          <Pressable
+            style={styles.editIconContainer}
+            onPress={handlePressEditIcon}
+          >
+            <Icon width={25} height={25} name="edit" color={'white'} />
+          </Pressable>
+        ),
     });
+
+    dispatch(coursesActions.getCategories());
   }, []);
 
+  useFocusEffect(
+    useCallback(() => {
+      return () => {
+        setEditMode(false);
+        reset({ newCategoryId: 0 });
```

Comment:
The default value if no category selected should be null
```suggestion
        reset({ newCategoryId: null });
```